core/store/models/common.go

```diff
@@ -731,6 +732,38 @@ func Merge(inputs ...JSON) (JSON, error) {
 	return JSON{Result: gjson.ParseBytes(bytes)}, nil
 }
 
+// MergeExceptResult does a merge, but will never clobber the field called "result"
+// On conflicting keys, rightmost inputs will clobber leftmost inputs EXCEPT if the field is named "result", in which case the leftmost result wins
+// This is needed to work around idiosyncrasies in the V1 job pipeline where "result" has special meaning
+func MergeExceptResult(inputs ...JSON) (JSON, error) {
+	output := make(map[string]interface{})
+
+	for _, input := range inputs {
+		switch v := input.Result.Value().(type) {
+		case map[string]interface{}:
+			for key, value := range v {
+				if key == "result" {
+					if _, exists := output["result"]; exists {
+						// Do not overwrite result field
```

Comment:
minor,  but maybe could print a warning if this happens? i suppose the original field name would need to be changed then. But it's v1, so...

Reply:
It's quite likely to occur often and not really an error in certain job specs, so I'd say not worth warning here.